File: SDKs/figshare.com/2.0.0/go/pkg/models/operations/privatearticlefileslist.go
package operations

import (
	"openapi/pkg/models/shared"
)

type PrivateArticleFilesListPathParams struct {
	ArticleID int64 `pathParam:"style=simple,explode=false,name=article_id"`
}

type PrivateArticleFilesListSecurity struct {
	OAuth2 shared.SchemeOAuth2 `security:"scheme,type=oauth2"`
}

type PrivateArticleFilesListRequest struct {
	PathParams PrivateArticleFilesListPathParams
	Security   PrivateArticleFilesListSecurity
}

type PrivateArticleFilesListResponse struct {
	ContentType  string
	ErrorMessage *shared.ErrorMessage
	PrivateFiles []shared.PrivateFile
	StatusCode   int64
}

File: SDKs/figshare.com/2.0.0/typescript/src/sdk/models/operations/privatearticlefileslist.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class PrivateArticleFilesListPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=article_id" })
  articleId: number;
}


export class PrivateArticleFilesListSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oAuth2: shared.SchemeOAuth2;
}


export class PrivateArticleFilesListRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PrivateArticleFilesListPathParams;

  @SpeakeasyMetadata()
  security: PrivateArticleFilesListSecurity;
}


export class PrivateArticleFilesListResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  errorMessage?: shared.ErrorMessage;

  @SpeakeasyMetadata({ elemType: shared.PrivateFile })
  privateFiles?: shared.PrivateFile[];

  @SpeakeasyMetadata()
  statusCode: number;
}
